src/main/java/org/openrewrite/staticanalysis/JavaElementFactory.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis;
+
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.Markers;
+
+import java.util.Collections;
+import java.util.Scanner;
+
+import static org.openrewrite.Tree.randomId;
+
+final class JavaElementFactory {
```

Comment:
Please note that this class was intentionally not made public. I don't know how we intend to address this kind of problem in the long run (creating LST elements which potentially refer to types from the source set). While I think it would be nice if `JavaTemplate` could be used for this, I have my doubts if we will be able to provide this any time soon. So in the meantime I suggest that we have a factory like this one.